go/app/main.go

```diff
@@ -1,79 +1,163 @@
+	Category  string `json:"category"`             // 商品のカテゴリ
+	ImageName string `json:"image_name,omitempty"` // 画像ファイル名
 }
 
-func root(c echo.Context) error {
-	res := Response{Message: "Hello, world!"}
-	return c.JSON(http.StatusOK, res)
+type Items struct {
+	Items []Item `json:"items"`
 }
 
-func addItem(c echo.Context) error {
-	// Get form data
-	name := c.FormValue("name")
-	c.Logger().Infof("Receive item: %s", name)
+func main() {
+	r := mux.NewRouter()
 
-	message := fmt.Sprintf("item received: %s", name)
-	res := Response{Message: message}
+	// 既存のエンドポイント
+	r.HandleFunc("/items", itemsHandler).Methods("POST", "GET")
 
-	return c.JSON(http.StatusOK, res)
+	// 新しいエンドポイント: 商品の詳細を取得
+	r.HandleFunc("/items/{item_id}", getItemHandler).Methods("GET")
+	r.HandleFunc("/search", searchItemsHandler).Methods("GET")
+
+	http.Handle("/", r)
+	log.Fatal(http.ListenAndServe(":8080", nil))
 }
 
-func getImg(c echo.Context) error {
-	// Create image path
-	imgPath := path.Join(ImgDir, c.Param("imageFilename"))
+func getItemHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	itemID := vars["item_id"]
 
-	if !strings.HasSuffix(imgPath, ".jpg") {
-		res := Response{Message: "Image path does not end with .jpg"}
-		return c.JSON(http.StatusBadRequest, res)
+	// items.json ファイルから商品情報を読み込む
+	var items Items
+	data, err := ioutil.ReadFile("items.json")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
 	}
-	if _, err := os.Stat(imgPath); err != nil {
-		c.Logger().Debugf("Image not found: %s", imgPath)
-		imgPath = path.Join(ImgDir, "default.jpg")
+	json.Unmarshal(data, &items)
+
+	// itemID に一致する商品を検索
+	for _, item := range items.Items {
+		if item.ID == itemID { // ID フィールドを Item 構造体に追加する必要があります
+			json.NewEncoder(w).Encode(item)
+			return
+		}
 	}
-	return c.File(imgPath)
-}
 
-func main() {
-	e := echo.New()
-
-	// Middleware
-	e.Use(middleware.Logger())
-	e.Use(middleware.Recover())
-	e.Logger.SetLevel(log.INFO)
+	// 商品が見つからない場合は 404 Not Found を返す
+	http.NotFound(w, r)
+}
 
-	frontURL := os.Getenv("FRONT_URL")
-	if frontURL == "" {
-		frontURL = "http://localhost:3000"
+func itemsHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "POST" {
+		// Multipart Formのパース
+		err := r.ParseMultipartForm(10 << 20) // 最大10MB
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		// JSONデータの処理
+		var item Item
+		item.Name = r.FormValue("name")
+		item.Category = r.FormValue("category")
+		// 商品情報にIDを割り当てる（ここでは現在のタイムスタンプを使用）
+		item.ID = strconv.FormatInt(time.Now().UnixNano(), 10)
+
+		// 画像の処理
+		file, header, err := r.FormFile("image")
+		if err != nil {
+			http.Error(w, "Image is required", http.StatusBadRequest)
+			return
+		}
+		defer file.Close()
+
+		// ハッシュ値を計算する
+		hash := sha256.New()
+		if _, err := io.Copy(hash, file); err != nil {
+			http.Error(w, "Failed to hash the image", http.StatusInternalServerError)
```

Comment:
ここはio.Copyに対するエラーなので、"Failed to hash the image"は少し違和感があります。以下のように書き換えもできます。

```suggestion
			http.Error(w, "Failed to copy the image", http.StatusInternalServerError)
```